models/articles.model.js

```diff
@@ -16,4 +19,21 @@ async function selectArticleById(id) {
     }
 }
 
-module.exports = {selectArticleById};
\ No newline at end of file
+async function selectArticles(sort_by='created_at') {
+    const validSortBys = ["created_at"];
+    if (!validSortBys.includes(sort_by)) {
+        return Promise.reject({status: 400, message: "bad request"})}
+    let sqlQueryString = '';
+    sqlQueryString += `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, CASE WHEN SUM(comments.votes) IS NULL THEN 0 
+    ELSE SUM(comments.votes)::INTEGER END AS votes, articles.article_img_url, COUNT(comments.article_id)::INTEGER AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id`
+    sqlQueryString += ` ORDER BY ${sort_by} DESC;`
+    try {
```

Comment:
Nice SQL going on here 👷🏻‍♀️

Reply:
Although it's not strictly necessary to do any string building or optional queries here, the endpoint should currently _always_ sort by date descending :)